tests/test_replicaset.py

```diff
@@ -144,23 +143,38 @@ def test_AutoReconnect(self):
             uri = "mongodb://localhost:{0}/?w={1}".format(self.ports[0], len(self.ports))
             conn = ConnectionPool(uri)
 
-            yield conn.db.coll.insert({'x': 42}, safe = True)
+            yield conn.db.coll.insert({'x': 42}, safe=True)
 
             yield self.__mongod[0].stop()
 
             while True:
                 try:
                     result = yield conn.db.coll.find_one()
+                    self.assertEqual(result['x'], 42)
                     break
                 except AutoReconnect:
                     pass
 
-            self.assertEqual(result['x'], 42)
         finally:
             yield conn.disconnect()
             self.flushLoggedErrors(AutoReconnect)
 
     @defer.inlineCallbacks
+    def test_AutoReconnect_from_primary_step_down(self):
+        if _version.version.major < 15:
+            raise unittest.SkipTest("No 'with self.assertRaises' in Twisted below version 15.")
+        self.patch(_Connection, 'maxDelay', 5)
+        uri = "mongodb://localhost:{0}/?w={1}".format(self.ports[0], len(self.ports))
+        conn = ConnectionPool(uri)
+
+        # this will force primary to step down, triggering an AutoReconnect that bubbles up
+        # through the connection pool to the client
+        with self.assertRaises(AutoReconnect):
+            yield conn.admin.command(SON([('replSetStepDown', 86400), ('force', 1)]))
```

Comment:
And sorry, I suggested you an ugly form of `command()`. It can be written nicier: `command("replSetStepDown", 86400, force=1)`